packages/user-management/src/common/core/email/SmtpEmailSender.ts

```diff
@@ -1,7 +1,92 @@
-import { Sender, MessageSender, Message, SentMessage } from '../../api';
+import { MessageSender, SentMessage, Email, EmailAddress } from '../../api';
+import { EmailError } from '../../api/exception/MessageError';
+import { EmailAddressWrapper, EmailContentWrapper, SmtpConfiguration, ConfigureSmtpTransport } from './email-utils';
+import { Transporter } from 'nodemailer';
+
+export class SmtpError extends EmailError {}
 
 export class SmtpEmailSender implements MessageSender {
-  async send(message: Message): Promise<SentMessage> {
-    throw 'Not implemented';
+  constructor(
+    private smtpConf: SmtpConfiguration,
+    private defaults: Partial<Email>,
+    private mailer: ConfigureSmtpTransport
+  ) {}
+
+  private transporter: Transporter;
+
+  async send(email: Email): Promise<SentMessage> {
+    try {
+      this.transporter = this.mailer({
+        host: this.smtpConf.host,
+        port: this.smtpConf.port,
+        secure: this.smtpConf.ssl,
+        tls: this.smtpConf.startTls,
+        auth: {
+          user: this.smtpConf.username,
+          pass: this.smtpConf.password
+        },
+        logger: false,
+        debug: false
+      });
+
+      const sentMessageByNodemailer = await this.transporter.sendMail(this.toSmtpMessage(email));
+
+      // Return the configuration if we are in a case of testing (mock)
+      if (sentMessageByNodemailer.response.includes('mock')) {
```

Comment:
This code must not be in the implementation class. Instead, the test should rprovide a mock instance of the mailer or transporter and the mock instance should be checked.
Never code specific parts for testing !